yt_dlp/extractor/neteasemusic.py

```diff
@@ -76,8 +76,8 @@ def _extract_formats(self, info):
         song_id = info['id']
         for level in self._LEVELS:
             song = traverse_obj(self._call_player_api(song_id, level), ('data', 0, {dict})) or {}
-            if song.get('level') != level:
-                break  # We have already extracted the highest level the user has access to
+            if song.get('level') in [song_format.get('level') for song_format in formats]:
+                break  # API returns existing level - the user have no access to further ones
```

Comment:
Maybe better to maintain a `set`, say `seen_levels`, and check for `song.get('level') in seen_levels`?

Reply:
Yes that's probably better, although I think [bashonly's commit](https://github.com/yt-dlp/yt-dlp/pull/10258/commits/009d2cc42f932e54554d02e54af2d4d325e383ad) should be enough